staging/src/k8s.io/code-generator/cmd/validation-gen/validators/common.go

```diff
@@ -40,16 +40,22 @@ func getMemberByJSON(t *types.Type, jsonName string) *types.Member {
 
 // isNilableType returns true if the argument type can be compared to nil.
 func isNilableType(t *types.Type) bool {
-	for t.Kind == types.Alias {
-		t = t.Underlying
-	}
-	switch t.Kind {
+	switch unaliasType(t).Kind {
 	case types.Pointer, types.Map, types.Slice, types.Interface: // Note: Arrays are not nilable
 		return true
 	}
 	return false
 }
 
+// unaliasType returns the underlying type of the specified type if it is an
+// alias, otherwise returns the type itself.
+func unaliasType(t *types.Type) *types.Type {
```

Comment:
Are we starting to accumulate a small collection of similar utilities? `deref`, `readType`, `unaliasType`?  Is there an opportunity to organize these in any way?

Reply:
Yeah, I actually want to revisit this.  I think I did this PR in a less-than-ideal way.  The more cases I find the clearer it becomes.  I'm going to put a hold on this one for the moment.

Reply:
PTAL - last commit does the retooling